src/util.cpp

```diff
@@ -328,8 +335,7 @@ const fs::path GetExportDir()
 
 const fs::path &GetDataDir(bool fNetSpecific)
 {
-
-    LOCK(csPathCached);
+    LOCK2(cs_args, csPathCached);
```

Comment:
Upstream seems to not have included the lock here; but it seems safe to.

Reply:
Yes, I think it's safe, but also necessary, because this function accesses `mapArgs` which is protected by `cs_args`. The upstream commit didn't change this function because it had already been updated to call `gArgs.GetArg()` (instead of accessing `mapArgs` directly), and `gArgs.GetArg()` calls `GetSetting()`, which acquires `cs_args`. We probably should pick up these argument-handling changes someday, but that's a big project.

Reply:
Yep, I've wanted to get these in for _years_.